server/libs/common/src/services/user.service.ts

```diff
@@ -0,0 +1,152 @@
+import { BadRequestException, ForbiddenException, Inject, Injectable, UnauthorizedException } from '@nestjs/common';
+import * as bcrypt from 'bcrypt';
+import { User } from '../models';
+
+export type UserCreateDto = Pick<User, 'email'> & Partial<User>;
+export type UserUpdateDto = Pick<User, 'id'> & Partial<User>;
+export interface UserCountDto {
+  admin?: boolean;
+}
+
+export interface IUserRepository {
```

Comment:
Would the interface for repository get move the its respective repostory?

Reply:
The service depends on this interface to work properly, so I think it makes sense that it stays here. You can have multiple implementation of this, which is one of the reasons for separating it. If we have two implementation (support for two different database backends, etc) they both should be importing the interface from the service, not the other way around.